src/lib/Libifl/pbsD_deljoblist.c

```diff
@@ -46,160 +46,471 @@
+		if (svrfd == iter_list->svr_fd) {
+			append_jobid(iter_list, job_id);
+			return 0;
+		}
+	}
+
+	new_node = calloc(1, sizeof(svr_jobid_list_t));
+	if (new_node == NULL) {
+		pbs_errno = PBSE_SYSTEM;
+		return 1;
+	}
+	new_node->svr_fd = svrfd;
+	append_jobid(new_node, job_id);
+	new_node->next = *svr_jobid_list_hd;
+	*svr_jobid_list_hd = new_node;
+
+	return 0;
 }
 
 /**
  * @brief
- *	-send Delete request and read reply.
+ *	Identify the respective svr_jobid_list struct
+ *  and calls the append_jobid function to append the jobid.
  *
- * @param[in] c - communication handle
- * @param[in] function - req type
- * @param[in] command - command
- * @param[in] objtype - object type
- * @param[in] objname - object name
- * @param[in] aoplp - attribute list
- * @param[in] extend - extend string for req
+ * @param[in] job_id - Job id
+ * @param[in] server - server name
+ * @param[in,out] svr_jobid_list_hd - head of the svr_jobib_list list
  *
- * @return	int
- * @retval	0	success
- * @retval	!0	error
+ * @return int
+ * @retval	0	- success
+ * @retval	1	- failure
  *
  */
-struct batch_deljob_status *
-PBSD_deljoblist(int c, int function, char **jobids, int numofjobs, char *extend)
+int
+add_jid_to_list_by_name(char *job_id, char *svrname, svr_jobid_list_t **svr_jobid_list_hd)
 {
-	int i;
-	struct batch_reply *reply;
-	struct batch_deljob_status *rbsp = NULL;
+	svr_jobid_list_t *iter_list = NULL;
+	svr_jobid_list_t *new_node = NULL;
 
-	/* initialize the thread context data, if not initialized */
-	if (pbs_client_thread_init_thread_context() != 0)
-		return NULL;
+	if ((job_id == NULL) || (svrname == NULL) || (svr_jobid_list_hd == NULL))
+		return 1;
 
-	/* lock pthread mutex here for this connection */
-	/* blocking call, waits for mutex release */
-	if (pbs_client_thread_lock_connection(c) != 0)
-		return NULL;
+	for (iter_list = *svr_jobid_list_hd; iter_list != NULL; iter_list = iter_list->next) {
+		if (strcmp(svrname, iter_list->svrname) == 0) {
+			append_jobid(iter_list, job_id);
+			return 0;
+		}
+	}
 
-	/* send the deletejoblist request */
-	i = PBSD_deljoblist_put(c, function, jobids, numofjobs, extend, PROT_TCP, NULL);
-	if (i) {
-		pbs_client_thread_unlock_connection(c);
-		return NULL;
+	new_node = calloc(1, sizeof(svr_jobid_list_t));
+	if (new_node == NULL) {
+		pbs_errno = PBSE_SYSTEM;
+		return 1;
+	}
+	new_node->svr_fd = -1;
+	pbs_strncpy(new_node->svrname, svrname, sizeof(new_node->svrname));
+	append_jobid(new_node, job_id);
+	new_node->next = *svr_jobid_list_hd;
+	*svr_jobid_list_hd = new_node;
+
+	return 0;
+}
+
+/**
+ * @brief	Helper function to add a new batch_deljob_status to the  list given
+ *
+ * @param[in]	jobid - id of the job
+ * @param[in]	code - error code
+ * @param[in]	list - list to add new object to
+ *
+ * @return	int
+ * @retval	0 for Success
+ * @retval	1 for Error
+ */
+static int
+add_failed_job(char *jobid, int code, struct batch_deljob_status **ret)
+{
+	struct batch_deljob_status *djstat = NULL;
+
```

Comment:
Please add a check for ret to be non-null

Reply:
for static functions I'm usually not as strict with such checks because it's kind of like an 'internal' function to that file, it exists for the sake of creating multiple smaller functions instead of one big function. But if you think it's important then I'll add it, please let me know.